pkg/kube/controllers/extendedjob/job_creator.go

```diff
@@ -46,9 +56,97 @@ type jobCreatorImpl struct {
 
 // Create satisfies the JobCreator interface. It creates a Job to complete ExJob. It returns the
 // retry if one of the references are not present.
-func (j jobCreatorImpl) Create(ctx context.Context, eJob ejv1.ExtendedJob) (retry bool, err error) {
+func (j jobCreatorImpl) Create(ctx context.Context, eJob ejv1.ExtendedJob, namespace string) (retry bool, err error) {
 	template := eJob.Spec.Template.DeepCopy()
 
+	// Create a service account for the pod
+	serviceAccount := &corev1.ServiceAccount{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      serviceAccountName,
+			Namespace: namespace,
+		},
+	}
+
+	// Bind read only role to the service account
+	roleBinding := &v1.RoleBinding{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "cluster-admin-role",
+			Namespace: namespace,
+		},
+		Subjects: []v1.Subject{
+			{
+				Kind:      "ServiceAccount",
+				Name:      serviceAccountName,
+				Namespace: namespace,
+			},
+		},
+		RoleRef: v1.RoleRef{
+			Kind:     "ClusterRole",
+			Name:     "cluster-admin",
+			APIGroup: "rbac.authorization.k8s.io",
+		},
+	}
+
+	// Set serviceaccount to the pod
+	template.Spec.ServiceAccountName = serviceAccountName
+
+	err = j.client.Create(ctx, serviceAccount)
+	if err != nil {
+		if !apierrors.IsAlreadyExists(err) {
+			return false, errors.Wrapf(err, "Could not create service account for pod in ejob %s.", eJob.Name)
```

Comment:
shouldn't capitalize first letter in an error

Reply:
Done.